aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -27,6 +27,8 @@
 void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha,
                     const Tensor& output_, std::string op_name, BinaryOpBlock binaryBlock)
 {
+  TORCH_CHECK(!(op_name == "power" && !is_macos_13_2_or_newer() && other.scalar_type() == ScalarType::Long && (self.scalar_type() != ScalarType::Half && self.scalar_type() != ScalarType::Float)),
```

Comment:
(self.scalar_type() != ScalarType::Half && self.scalar_type() != ScalarType::Float) is excluded because square f16 and f32 works, power (float, exp = (Long)2) will promote int64 to float